File: article_35/07_string_comparison.c
#include <stdio.h>
#include <string.h>

int main(void) {
	printf("%d\n", strcmp("aa", "ab"));
	printf("%d\n", strcmp("aa", "aa"));

	printf("%d\n", strcmp("e", "é"));
	printf("%d\n", strcmp("e", "ě"));
	printf("%d\n", strcmp("ě", "é"));

	printf("%d\n", strcmp("z", "é"));
	printf("%d\n", strcmp("z", "ě"));

	printf("%d\n", strcmp("h", "ch"));
	printf("%d\n", strcmp("ch", "i"));

	printf("%d\n", strcmp("Hrdina", "Chocholoušek"));
	printf("%d\n", strcmp("Crha", "Chocholoušek"));

        return 0;
}

File: article_35/07_string_comparison.go
// Seriál "Programovací jazyk Go"
//    https://www.root.cz/serialy/programovaci-jazyk-go/
//
// Třicátá pátá část
//    Programovací jazyk Go pro skalní céčkaře (2.část)
//    https://www.root.cz/clanky/programovaci-jazyk-go-pro-skalni-ceckare-2-cast/
//
// Seznam příkladů ze třicáté páté části:
//    https://github.com/tisnik/go-root/blob/master/article_35/README.md
//
// Demonstrační příklad číslo 7:
//     Porovnávání řetězců v Go.

package main

import "fmt"

func main() {
	fmt.Println("aa" < "ab")
	fmt.Println("aa" == "ab")

	fmt.Println("aa" < "aa")
	fmt.Println("aa" == "aa")

	fmt.Println("e" < "é")
	fmt.Println("e" < "ě")

	fmt.Println("z" < "é")
	fmt.Println("z" < "ě")

	fmt.Println("h" < "ch")
	fmt.Println("ch" < "i")

	fmt.Println("Hrdina" < "Chocholoušek")
	fmt.Println("Crha" < "Chocholoušek")
}
